lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -0,0 +1,37 @@
+module PulpProxy
+  class DiskUsage
+    include ::Proxy::Util
+    SIZE = {:kilobyte => 1_024, :megabyte => 1_048_576, :gigabyte => 1_073_741_824, :terabyte => 1_099_511_627_776}
+
+    attr_reader :path, :stat, :size
+
+    def initialize(opts ={})
+      @path = opts[:path] || PulpProxy::Plugin.settings.pulp_dir
+      @size = SIZE[opts[:size]] || SIZE[:gigabyte]
+      get_stat
+    end
+
+    def to_json
+      data.to_json
+    end
+
+    def data
+      stat.merge({:path => path, :size => SIZE.key(size)})
+    end
+
+    private
+
+
+    def get_stat
+      command_path = which("df")
```

Comment:
@witlessbird is there a way to fail the plugin if we cant find a binary? imho this whole plugin should not load if the df command is not found.